### PR TITLE
Separating db engine and core

### PR DESCRIPTION
**Pull request by benedeki**
* Separated the core and the Slick implementation
* Core is now independent and type-parametrized
* Enceladus moved to examples folder
* Deleted all kind unneeded code

### CODE REVIEW

**Comment by benedeki** [comment]
This my be easier to review as the whole code rather than a PR with code compare. /methinks

**Review by dk1844 — APPROVED** [review]
LGTM (just read the code). To the degree that I have understood it, I see nothing wrong.

File: src/main/scala/za/co/absa/fadb/naming_conventions/lettersCase/LettersCase.scala [hunk 1]
```diff
@@ -13,7 +13,7 @@
  * limitations under the License.
  */
 
-package za.co.absa.faDB.namingConventions.lettersCase
+package za.co.absa.fadb.naming_conventions.lettersCase
```
Comment (AdrianOlosutean) [inline src/main/scala/za/co/absa/fadb/naming_conventions/lettersCase/LettersCase.scala:16]: Probably `lettersCase` package should also be renamed

**Review by dk1844 — APPROVED** [review]
LGTM (just read the code)

**Review by AdrianOlosutean — APPROVED** [review]
Looks good

**Pull request merged**